internal/convenience/synchronizer/graphile_synchornizer_test.go

```diff
@@ -0,0 +1,204 @@
+package synchronizer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/calindra/nonodo/internal/convenience/model"
+	"github.com/calindra/nonodo/internal/convenience/repository"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type DecoderInterfaceMock struct {
+	mock.Mock
+}
+
+func (m *DecoderInterfaceMock) RetrieveDestination(output model.OutputEdge) (common.Address, error) {
+	args := m.Called(output)
+	return args.Get(0).(common.Address), args.Error(1)
+}
+
+func (m *DecoderInterfaceMock) GetConvertedInput(input model.InputEdge) (model.ConvertedInput, error) {
+	args := m.Called(input)
+	return args.Get(0).(model.ConvertedInput), args.Error(1)
+}
+
+func (m *DecoderInterfaceMock) HandleOutput(ctx context.Context, destination common.Address, payload string, inputIndex uint64, outputIndex uint64) error {
+	args := m.Called(ctx, destination, payload, inputIndex, outputIndex)
+	return args.Error(0)
+}
+
+func (m *DecoderInterfaceMock) HandleInput(ctx context.Context, input model.InputEdge, status model.CompletionStatus) error {
+	args := m.Called(ctx, input, status)
+	return args.Error(0)
+}
+
+func (m *DecoderInterfaceMock) HandleReport(ctx context.Context, index int, outputIndex int, payload string) error {
+	args := m.Called(ctx, index, outputIndex, payload)
+	return args.Error(0)
+}
+
+func getTestOutputResponse() OutputResponse {
+	jsonData := `
```

Comment:
How did you get this graphile response?

Reply:
I generated this JSON according to the OutputResponse model.